Quotient/connection.cpp

```diff
@@ -211,10 +211,7 @@ void Connection::assumeIdentity(const QString& mxId, const QString& accessToken)
             d->completeSetup(job->userId());
         });
         connect(job, &BaseJob::failure, this, [this, job] {
-            if (job->error() == BaseJob::StatusCode::NetworkError)
-                emit networkError(job->errorString(), job->rawDataSample(),
-                                  job->maxRetries(), -1);
-            else
+            if (job->error() != BaseJob::StatusCode::NetworkError)
```

Comment:
You're effectively reverting e9c3d3799 here, which reopens #614. `networkError()` is emitted here with `-1` for the time until the next attempt to signify that the job has run out of attempts (probably it should be documented). Quaternion relies on this to retry `assumeIdentity()` indefinitely; but now that I think of it, it's probably reasonable to just make `assumeIdentity()` try indefinitely instead, the same way `SyncJob` does. In any case, that piece has to be either reverted, or overhauled.

Reply:
When you mean "that piece", are you referring to the magic `-1` handling? Would you suggest modifying `assumeIdentity()` to have that behavior should happen here in quotient or in quaternon?

Reply:
I mean the lambda code here. I think it would be ideal to call `setMaxRetries()` on this particular job instance. It's just a bit complicated but not much.

`GetTokenOwnerJob` itself is generated from the API definition, you can't change it directly - and actually, you shouldn't because it would affect all respective requests, including those from the client code. As of now, `setMaxRetries()` is protected but there's actually no particular reason it should be this way. So what I suggest is:
1. Make `BaseJob::setMaxRetries()` public. As far as I can tell, it doesn't even break the ABI (and definitely doesn't break the API) compatibility.
2. Call `job->setMaxRetries()` on the job object returned by `callApi()`. This is safe to do at any point in the job's lifecycle, no need to delay the job execution.
3. Delete emission of `networkError()` (you already did it here).